driver-mapping/src/main/java/com/datastax/driver/mapping/MapperConfiguration.java

```diff
@@ -0,0 +1,336 @@
+ * </ul>
+ * This is also where you get the configured settings, though those cannot be changed
+ * (they are set during the built of the Mapper object).
+ */
+public class MapperConfiguration {
+
+    private PropertyScanConfiguration propertyScanConfiguration;
+
+    public MapperConfiguration() {
+        this.propertyScanConfiguration = new PropertyScanConfiguration();
+    }
+
+    public MapperConfiguration(MapperConfiguration toCopy) {
+        this.propertyScanConfiguration = new PropertyScanConfiguration(toCopy.propertyScanConfiguration);
+    }
+
+    /**
+     * Returns the property scanning configuration
+     *
+     * @return the property scanning configuration
+     */
+    public PropertyScanConfiguration getPropertyScanConfiguration() {
+        return propertyScanConfiguration;
+    }
+
+    /**
+     * Sets the property scanning configuration
+     *
+     * @param propertyScanConfiguration property scanning configuration to use
+     * @return the MapperConfiguration to enable builder pattern
+     */
+    public MapperConfiguration setPropertyScanConfiguration(PropertyScanConfiguration propertyScanConfiguration) {
+        this.propertyScanConfiguration = propertyScanConfiguration;
+        return this;
+    }
+
+    public static class PropertyScanConfiguration {
+
+        private Collection<String> excludedProperties;
+
+        private PropertyScanScope propertyScanScope;
+
+        private PropertyMappingStrategy propertyMappingStrategy;
+
+        private HierarchyScanStrategy hierarchyScanStrategy;
+
+        public PropertyScanConfiguration() {
+            this.excludedProperties = new HashSet<String>();
+            this.propertyScanScope = new PropertyScanScope();
+            this.propertyMappingStrategy = PropertyMappingStrategy.BLACK_LIST;
+            this.hierarchyScanStrategy = new HierarchyScanStrategy();
+        }
+
+        public PropertyScanConfiguration(PropertyScanConfiguration toCopy) {
+            this.excludedProperties = toCopy.excludedProperties;
+            this.propertyScanScope = new PropertyScanScope(toCopy.propertyScanScope);
+            this.propertyMappingStrategy = toCopy.propertyMappingStrategy;
+            this.hierarchyScanStrategy = new HierarchyScanStrategy(toCopy.hierarchyScanStrategy);
+        }
+
+        /**
+         * Returns a collection of properties to exclude from mapping
+         *
+         * @return a collection of properties to exclude from mapping
+         */
+        public Collection<String> getExcludedProperties() {
+            return excludedProperties;
+        }
+
+        /**
+         * Sets a collection of properties to exclude from mapping
+         *
+         * @param excludedProperties a collection of properties to exclude from mapping
+         * @return the PropertyScanConfiguration to enable builder pattern
+         */
+        public PropertyScanConfiguration setExcludedProperties(Collection<String> excludedProperties) {
```

Comment:
Related to [Olivier's comment about immutability](https://github.com/olim7t), Unless I missed a previous discussion rather than setters, I would prefer we use a builder strategy for `PropertyScanConfiguration` and `MappingConfiguraiton`.  Once intialized it would be nice if the configuration would not change and we use a builder strategy for a lot of other modes of configuration.